File: appejv-api/internal/models/order.go
package models

import "time"

type Order struct {
	ID          int        `json:"id"`
	CustomerID  int        `json:"customer_id"`
	SaleID      string     `json:"sale_id"`
	Status      string     `json:"status"`
	TotalAmount float64    `json:"total_amount"`
	DeletedAt   *time.Time `json:"deleted_at,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
}

type OrderItem struct {
	ID           int     `json:"id"`
	OrderID      int     `json:"order_id"`
	ProductID    int     `json:"product_id"`
	Quantity     int     `json:"quantity"`
	PriceAtOrder float64 `json:"price_at_order"`
}

type CreateOrderRequest struct {
	CustomerID int               `json:"customer_id" binding:"required"`
	Items      []OrderItemCreate `json:"items" binding:"required,min=1"`
}

type OrderItemCreate struct {
	ProductID int `json:"product_id" binding:"required"`
	Quantity  int `json:"quantity" binding:"required,min=1"`
}

type UpdateOrderRequest struct {
	Status *string `json:"status"`
}

type OrderWithDetails struct {
	Order
	Customer     Customer      `json:"customer"`
	Items        []OrderItem   `json:"items"`
	ItemProducts []Product     `json:"item_products"`
}

File: appejv-android/app/src/main/java/com/appejv/android/core/data/model/Order.kt
package com.appejv.android.core.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Order(
    @SerialName("id")
    val id: Int,
    
    @SerialName("customer_id")
    val customerId: String? = null,
    
    @SerialName("sale_id")
    val saleId: String,
    
    @SerialName("status")
    val status: String,
    
    @SerialName("total_amount")
    val totalAmount: Double,
    
    @SerialName("created_at")
    val createdAt: String,
    
    @SerialName("updated_at")
    val updatedAt: String? = null,
    
    @SerialName("items")
    val items: List<OrderItem>? = null
)

@Serializable
data class OrderItem(
    @SerialName("id")
    val id: Int,
    
    @SerialName("order_id")
    val orderId: Int,
    
    @SerialName("product_id")
    val productId: Int,
    
    @SerialName("quantity")
    val quantity: Int,
    
    @SerialName("price_at_order")
    val priceAtOrder: Double,
    
    @SerialName("product")
    val product: Product? = null
)

@Serializable
data class OrdersResponse(
    @SerialName("data")
    val data: List<Order>,
    
    @SerialName("total")
    val total: Int? = null
)

@Serializable
data class CreateOrderRequest(
    @SerialName("customer_id")
    val customerId: String? = null,
    
    @SerialName("items")
    val items: List<CreateOrderItem>
)

@Serializable
data class CreateOrderItem(
    @SerialName("product_id")
    val productId: Int,
    
    @SerialName("quantity")
    val quantity: Int,
    
    @SerialName("price_at_order")
    val priceAtOrder: Double
)
